### PR TITLE
Migration to add domain field for mobile endpoint restriction setting (1 of 2 PRs)

### PR DESCRIPTION
**Pull request by AddisonDunn**
## Product Description

This is part of a push to improve security around mobile endpoints. [Here is the description](https://docs.google.com/document/d/1aVDHalKJvtTc8FVkMmFb0fxzxO5UaW2-_4g2MTQlrz0/edit?usp=sharing) of the vulnerabilities and proposed changes written up by Ethan. Here is the [Jira epic](https://dimagi-dev.atlassian.net/browse/USH-1408).

The feature flag RESTRICT_MOBILE_ENDPOINTS currently restricts mobile workers' access to the mobile app (and thus the mobile endpoints) by allowing administrators to it turn on/off on the Roles & Permissions page.

 **This PR is one of 2 PRs that will cause that feature flag to become just a switch to display a new GA setting**. 

## Technical Summary

[USH-1409](https://dimagi-dev.atlassian.net/browse/USH-1409).

This preliminary migration will add a new field to the Domain model and set this field to True if the relevant feature flag is turned on. The hope is that this can be merged before tonight's deploy so that we can see that the migration has been applied before adding the second PR next week.

This is step 1. Step 2 will include UI changes and a transition from the existing FF to the GA setting.

## Feature Flag

RESTRICT_MOBILE_ACCESS

### Safety story

I tested that this properly works locally and on staging (domain.restrict_mobile_access = true when the FF is on, false if not). No major errors in the process.

### Automated test coverage

No relevant tests I could find.

### QA Plan

None planned for this migration _or_ the next PR.

### Migrations

- [x] The migrations in this code can be safely applied first independently of the code

Code that depends on this migration will be deployed in the following PR.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/domain/migrations/0009_restrict_mob_access_from_FF.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+from corehq.util.django_migrations import skip_on_fresh_install
+
+from corehq.toggles import RESTRICT_MOBILE_ACCESS
+
+
+@skip_on_fresh_install
+def _enable_restrict_mobile_access(apps, schema_editor):
+    for domain in RESTRICT_MOBILE_ACCESS.get_enabled_domains():
+        domain_obj = Domain.get_by_name(domain)
+        domain_obj.restrict_mobile_access = True
```
[the remark this answers is not in the document]
Comment (AddisonDunn) [reply corehq/apps/domain/migrations/0009_restrict_mob_access_from_FF.py]: I addressed both of these items in my new commit. Re-tested the migration locally and with an intentionally misspelled domain name in the FF page.

**Review by orangejenny — APPROVED** [review]
looks great

**Review by snopoke — COMMENTED** [review]
Just confirming that we'll need to run a migration like this again with the code change to use this new field (in case there are new domains added to the FF)

**Comment by AddisonDunn** [comment]
I didn't think about that. @orangejenny I think Simon is right: if a domain is added to the FF in the meantime, the setting won't be turned on when the next PR is deployed.

What's the best way to re-run the Django migration before deploying the second PR? I can think of the private release option or creating another migration file but neither of those sounds ideal.

**Comment by snopoke** [comment]
You'll need to extract the logic into a management command which can be run via the shell.

**Comment by orangejenny** [comment]
@snopoke @AddisonDunn I admit, I was thinking that Addison could just verify pre-deploy that all of the domains with the flag on also have the new domain attr turned on. That isn't the most robust approach in general, but this is a custom flag and is USH-owned, so the set of domains using the flag should be quite stable.

**Comment by AddisonDunn** [comment]
@orangejenny I remember you mentioning that now. If they are that stable then I think it makes plenty of sense for me to just check beforehand since it's something I already know how to do. I guess I'd just check late in the workday Wednesday before the deploy that evening?

Then I can check again on Thursday after deploy to make sure bychance no new domain was added in that short time period.

**Comment by orangejenny** [comment]
@AddisonDunn yeah, that's what I was imagining

**Comment by AddisonDunn** [comment]
OK then unless you think this isn't sufficient @snopoke I'll make that my plan. Thanks regardless for helping clear up this important point.

**Pull request merged**